### PR TITLE
ContentTyped Improvements

### PR DESCRIPTION
**Pull request by ryneeverett**
The relevant conversation is so far all in stephenmcd/cartridge/pull/293.

### CODE REVIEW

**Comment by ryneeverett** [comment]
Let me know if those tests are garbage. I'm sort of clueless when it comes to writing effective tests from within django's test <s>straightjacket</s> harness.

**Comment by stephenmcd** [comment]
Tests look fine, thanks Ryne.

File: mezzanine/core/models.py [hunk 1]
```diff
@@ -524,8 +529,22 @@ def get_content_models(cls):
...
+        self.content_model = (
+            None if is_base_class else self.get_content_model_name())
+
     def get_content_model(self):
-        return getattr(self, self.content_model, None)
+        """
+        Return content model, or if this is the base class return it.
+        """
+        return (getattr(self, self.content_model) if self.content_model
+                else self.__class__)
```
Comment (AlexHill) [inline mezzanine/core/models.py]: This should be `self` rather than `self.__class__` – `get_content_model()` returns an instance.

Comment (AlexHill) [inline mezzanine/core/models.py]: Fixed in 3cc2e7a8b73cdafbb231853386d049ac669bcaf7

Comment (stephenmcd) [inline mezzanine/core/models.py]: Thanks Alex!

**Comment by AlexHill** [comment]
Thanks Ryne, this is great. Let's keep this ball rolling and get the Cartridge changes merged ASAP.

File: mezzanine/core/models.py (hunk 1 above)
Comment (ryneeverett) [inline mezzanine/core/models.py]: whoops! Thanks for fixing it.

**Comment by AlexHill** [comment]
As a sub-project, I'm going to go through Mezzanine to make sure every usage of `content_model` or `get_content_model()` has test coverage.

**Comment by AlexHill** [comment]
I've created a branch for that at https://github.com/stephenmcd/mezzanine/compare/content_typed_tests

**Pull request merged**